src/app/Plugins/Calendar/Components/new-event/new-event.component.ts

```diff
@@ -0,0 +1,232 @@
+import { Component, OnInit, ViewEncapsulation, ViewChild } from '@angular/core';
+import { CalendarEvent } from '../../Objects/CalendarEvent';
+import { CalendarDataService } from '../../Services/calendar-data.service';
+import { LocalDateTime } from '../../../../Objects/LocalDateTime';
+import { MatInput } from '@angular/material';
+
+/*
+  Component for setting up a new event.
+
+  @author: Florian Bauer
+*/
+@Component({
+  selector: 'app-new-event',
+  templateUrl: './new-event.component.html',
+  styleUrls: ['./new-event.component.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class NewEventComponent implements OnInit {
+
+  title: string;
+  description: string;
+  allDay: boolean;
+  address: string;
+  city: string;
+  country: string;
+  postalCode: string;
+  location: string;
+  reminderTime: number;
+  timeUnit: string;
+  reminderType: string;
+  titleChoosen: boolean;
+  startChoosen: boolean;
+  endChoosen: boolean;
+  minDate: Date;
+  endDate: Date;
+  startTime2: string;
+  endTime2: string;
+  hintString: string;
+  missingInformation: string[];
+  correctTime: boolean;
+
+  @ViewChild('eventStart', {
+    read: MatInput
+  }) eventStart: MatInput;
+
+  @ViewChild('eventEnd', {
+    read: MatInput
+  }) eventEnd: MatInput;
+
+  constructor(private readonly calendarService: CalendarDataService) { }
+
+  ngOnInit() {
+    this.resetValues();
+  }
+
+  setStart(): void {
+    if (this.startTime2 === '') {
+      this.startTime2 = '00:00';
+    }
+  }
+
+  setTitle(): void {
+    if (this.title !== '') {
+      this.titleChoosen = true;
+    } else {
+      this.titleChoosen = false;
+      this.startChoosen = false;
+      this.endChoosen = false;
+    }
+  }
+
+  setEnd(): void {
+    if (this.endTime2 === '') {
+      this.endTime2 = '23:59';
+    }
+  }
+
+  setAllDay() {
+    this.allDay = !this.allDay;
+  }
+
+  startDateChoosen(dateValue): void {
+    this.startChoosen = true;
+    const makeDate = new Date(dateValue);
+    this.minDate = new Date(makeDate.getFullYear(), makeDate.getMonth(), makeDate.getDate());
```

Comment:
what is this for? Why not just use `makeDate`? If this is intentional, add a comment and explain why

Reply:
I thought that the timestamp could be a problem, but now that you ask it could be obsolete